src/ariel-os-hal/src/dummy/mod.rs

```diff
@@ -13,6 +13,7 @@
     reason = "this dummy module mimics manufacturer-specific crates"
 )]
 
+#[cfg(not(context = "native"))]
```

Comment:
Just three special cases in the dummy module allows using it for native w/o warnings.
`ariel-os-native` is basically empty now. I kept it so we have the scaffolding to add impls there.